sympy/printing/codeprinter.py

```diff
@@ -78,23 +78,27 @@ def doprint(self, expr, assign_to=None):
         expr : Expression
             The expression to be printed.
 
-        assign_to : Symbol, MatrixSymbol, or string (optional)
-            If provided, the printed code will set the expression to a
-            variable with name ``assign_to``.
+        assign_to : Symbol, string, MatrixSymbol, list of strings or Symbols (optional)
+            If provided, the printed code will set the expression to a variable or multiple variables
+            with the name or names given in ``assign_to``.
         """
-        from sympy.codegen.ast import Assignment
         from sympy.matrices.expressions.matexpr import MatrixSymbol
+        from sympy.codegen.ast import CodeBlock, Assignment
 
         if isinstance(assign_to, str):
             if expr.is_Matrix:
                 assign_to = MatrixSymbol(assign_to, *expr.shape)
             else:
                 assign_to = Symbol(assign_to)
+        elif isinstance(assign_to, (list, tuple)):
+            assign_to = tuple(x if isinstance(x, Symbol) else Symbol(x) for x in assign_to)
         elif not isinstance(assign_to, (Basic, type(None))):
             raise TypeError("{0} cannot assign to object of type {1}".format(
                     type(self).__name__, type(assign_to)))
```

Comment:
It strikes me as odd that the str -> symbol logic is not the same for lists and single elements. Perhaps something like:
```python
def _handle_assign_to(expr, assign_to):
    if assign_to is None:
        return sympify(expr)
    if isinstance(assign_to, (list, tuple)):
        if len(expr) != len(assign_to):
            raise ValueError(...)
        return CodeBlock([_handle_assign_to(rhs, lhs) for lhs, rhs in zip(expr, assign_to)
    if isinstance(assign_to, str):
        assign_to = ... # as before
    elif not isinstance(assign_to, Basic):
        raise ... # as before
    return Assignment(assign_to, expr)

expr = _handle_assign_to(expr, assign_to)
```
Which would _maybe_ support nested tuple targets too.

Reply:
Okay!  Nice suggestion, I implemented this in the last commit, and basic nested tuple/list assignments work just as you'd imagined :partying_face:.  It was simplest to make a test case in the `test_glsl.py` file, since `codeprinter.py` is pretty abstract and doesn't implement all the basic printing functions.  Hopefully that's alright.